pkg/ccl/changefeedccl/kvfeed/kv_feed.go

```diff
@@ -592,11 +618,15 @@ func (f *kvFeed) runUntilTableEvent(ctx context.Context, resumeFrontier span.Fro
 	}()
 
 	// We have catchup scan checkpoint.  Advance frontier.
-	if startFrom.Less(f.checkpointTimestamp) {
-		for _, s := range f.checkpoint {
-			if _, err := resumeFrontier.Forward(s, f.checkpointTimestamp); err != nil {
-				return err
-			}
+	if f.spanLevelCheckpoint != nil || startFrom.Less(f.checkpointTimestamp) {
```

Comment:
can you add an issue to remove this conditional check, since frontier.Forward cannot regress. and add a TODO(#issue) here

Reply:
Done.